core/lib/src/config/secret_key.rs

```diff
@@ -1,15 +1,15 @@
 use std::fmt;
 
-use aes_gcm::{
-    AeadCore, Aes256Gcm, Nonce,
-    aead::{generic_array::GenericArray, Aead, KeyInit, OsRng},
+use chacha20poly1305::{
+    aead::{Aead, AeadCore, KeyInit, OsRng, generic_array::GenericArray},
+    XChaCha20Poly1305, XNonce,
 };
 use cookie::Key;
 use serde::{de, ser, Deserialize, Serialize};
 
 use crate::request::{Outcome, Request, FromRequest};
 
-const NONCE_LEN: usize = 12;
+const NONCE_LEN: usize = 24; // 192-bit
```

Comment:
I don't think we need this anymore. There's [`NonceSize`](https://docs.rs/chacha20poly1305/latest/chacha20poly1305/trait.AeadCore.html#associatedtype.NonceSize).